### ISSUE TITLE
Removing the separate email notification for money requests and refactoring unread message summary to include it instead

### ISSUE THREAD

**Issue by anmurali** [Expensify/App#34167]
Context [here](https://expensify.slack.com/archives/C05NJ4SLBMF/p1704221739796809)
**Summarizing what we want with email notifications for money requests:**

1. We want to remove the separate email notification that goes out to users when they receive a money request. Rather, we want to fold that into the existing `unreadMessageSummary` construct
2. We still want to notify the user immediately via email when they receive a money request
3. We want to customize the subj line of the unreadMessageSummary email to make it clear they have a money request
4. We want this subj line to be satisfying, i.e. it should ideally say who is asking you for what
5. We want this email notification body to look exactly like the DM (or workspace chat, or group chat..) in product
6. We want any CTAs to automatically bring the user logged in to the action

**Comment by anmurali** [comment]
I will work on mocking this up first with @shawnborton and then we can apply the `engineering` label

**Comment by shawnborton** [comment]
Not overdue, on my plate for this week.

**Comment by shawnborton** [comment]
@anmurali is there any more context that you are planning to provide before I dig it? Just responding to your comment "I will work on mocking this up first"

**Comment by anmurali** [comment]
@shawnborton sorry I have not had a chance to jump in here since last week but I will provide more context on this tomorrow. This is also categorized as `Low`, so I am going to switch it to a `Weekly`

**Comment by shawnborton** [comment]
Sounds good, just keep me posted!

**Comment by anmurali** [comment]
Ok, so the scope of this issue is just to refactor the email notification that goes out when a money request is sent (DM or workspace chat). 

**This is what it looks like today**
<img width="1074" alt="image" src="https://github.com/Expensify/App/assets/14225673/55636751-0dd0-4a0d-8c22-1f35ca591e23">

What we want to do is reuse the `unreadMessage` notification, specifically when custom subject lines are used. 

This is what an email notification for a message that uses custom subject line looks like:

**Chat view (Highlighted message):**
<img width="1079" alt="image" src="https://github.com/Expensify/App/assets/14225673/9306d14d-9f13-44ab-91bc-6825215c67a6">

**Email view**
<img width="1073" alt="image" src="https://github.com/Expensify/App/assets/14225673/c8e22efd-df04-4872-8d55-0999949a9fb1">

To reuse this construct for money request notifications:

Title/ Subj: Sender requests $amount 
_Use Sender first and last name if available, fall back to sender email or phone number if not available_

Body: We want this to be HTML and look exactly like the money request in the DM
<img width="1077" alt="image" src="https://github.com/Expensify/App/assets/14225673/fa3ba7b3-8e1a-4a46-9872-91bb8b61df86">

Footer: This message was sent from Expensify. You can reply directly to this message, or pay by clicking on the `Pay` button

**Comment by shawnborton** [comment]
> and how the money request will look using custom subj line in the chat itself?

This part tripped me up a bit. Does this mean we'd be adding a custom subject line to every single money request in the app?

**Comment by shawnborton** [comment]
To get the conversation started though, here is a first stab at this:
![image](https://github.com/Expensify/App/assets/2319350/870dda06-baa3-4d87-b085-04a48d91bf98)


- The email comes from Expensify (notifications@expensify.com)
- The body of the email looks pretty close to a chat with the actual chat UI
- Footer message included as you requested

We might also consider doing something like a small intro blurb to our emails, something like: 
![image](https://github.com/Expensify/App/assets/2319350/3eb15aaa-0f33-4295-8aec-7c716bdeea7f)

or even add more logo power:
![image](https://github.com/Expensify/App/assets/2319350/55c50b94-cbed-461e-bf85-04d7dbe4f4fe)


Then we can get even crazier and consider how we might just group together multiple outstanding items that need your attention:
![image](https://github.com/Expensify/App/assets/2319350/40691283-5628-4049-97d4-144f0868fe41)


(relevant thread in Slack about that idea above [here](https://expensify.slack.com/archives/C05NJ4SLBMF/p1705345695801229), cc @RobertLadue @LLPeckham)

**Comment by anmurali** [comment]
>This part tripped me up a bit. Does this mean we'd be adding a custom subject line to every single money request in the app?

Yes, that's the idea. So maybe we should also add a mockup that shows how that looks in product?

>Then we can get even crazier and consider how we might just group together multiple outstanding items that need your attention:

This will be the actual unread message summary. The custom subject line kicks off a realtime email notification of that specific message, which is the money request and nothing but the money request. But the general unread message summary, which is still sent when you're not active in the product would combine all the outstanding items (chats/ requests and everything else you missed)

I personally like the intro blurb _without_ the logo power but if you look at the one with the logo power, it says Expensify so many times that it looks weird! What do you think @shawnborton 
<img width="400" alt="image" src="https://github.com/Expensify/App/assets/14225673/a86a35de-3346-4c67-9b4a-eb2090a63c26">

Speaking of intro blurb, should we also include a subtext to say what our product is/ does? Like Venmo does?
<img width="400" alt="image" src="https://github.com/Expensify/App/assets/14225673/a1612924-5ab5-4623-84c9-dc305687b380">
They have a lot, we needn't add ALL that - they do cause their MTLs require them to I think. But just a single sentence saying what our product is might be good? cc @GabiHExpensify

**Comment by shawnborton** [comment]
I really don't think we should start adding subject lines above money requests - it definitely makes that flow feel way heavier. I think it's fine if the email subject exists but a subject doesn't exist in the product flow. Curious if @Expensify/design has any thoughts here too.

![CleanShot 2024-01-21 at 19 16 31@2x](https://github.com/Expensify/App/assets/2319350/9632038a-870d-4cc5-9fcc-607694c81813)

**Comment by shawnborton** [comment]
> I personally like the intro blurb without the logo power but if you look at the one with the logo power, it says Expensify so many times that it looks weird! What do you think @shawnborton

I agree with this in this particular case. So maybe something like this feels nicest?
![image](https://github.com/Expensify/App/assets/2319350/e00ea039-df62-4887-8027-32019116fc7f)

Though I wonder if we want to see how all of this will interact with the work that Lindsey and Bob are doing with the combined daily summary? It feels like there is some overlap here.

**Comment by shawnborton** [comment]
> Speaking of intro blurb, should we also include a subtext to say what our product is/ does? Like Venmo does?

I don't feel too strongly here, but I can see where this is beneficial for users who have never heard of Expensify before and are receiving a request for the first time. I wonder if we could detect that in the notification and add this blurb for when we send something to a user who has never interacted with Expensify before?

**Comment by dubielzyk-expensify** [comment]
>I really don't think we should start adding subject lines above money requests - it definitely makes that flow feel way heavier. I think it's fine if the email subject exists but a subject doesn't exist in the product flow. Curious if @Expensify/design has any thoughts here too.

I agree with this

**Comment by dannymcclain** [comment]
☝️ Same.

**Comment by anmurali** [comment]
Don't we NEED that custom subject line above the money request to leverage the custom subject line logic on unreadMessage summary? I am wondering if your design feedback effectively means back to the drawing board on the plan itself https://github.com/Expensify/App/issues/34167#issue-2072826021, especially 1 and 2?

**Comment by shawnborton** [comment]
> Don't we NEED that custom subject line above the money request to leverage the custom subject line logic on unreadMessage summary?

Can't we just send whatever we want as an email subject? Like right now we are just choosing a subject that doesn't come from the chat messages right? 

> I am wondering if your design feedback effectively means back to the drawing board on the plan itself https://github.com/Expensify/App/issues/34167#issue-2072826021, especially 1 and 2?

I'm not entirely sure what you mean by this. I do think we can:
- make the email look more like the real chat, by including the preview card
- send whatever email subject we think is best in this particular case, even if the message itself doesn't have a subject

**Comment by anmurali** [comment]
Bringing this discussion to Slack [here](https://expensify.slack.com/archives/C05NJ4SLBMF/p1705963209954789)

**Comment by anmurali** [comment]
Ok based on that discussion above, I updated the issue description. Final mockup:

![image](https://github.com/Expensify/App/assets/14225673/9646228e-4411-4830-9888-b153b7d0c870)

**Comment by shawnborton** [comment]
Sounds good. Do we want to first just make the email behave correctly in terms of text/content, and then follow up with style? Mostly just saying that in reference to David's comment [here](https://expensify.slack.com/archives/C05NJ4SLBMF/p1706131611646059?thread_ts=1705345695.801229&cid=C05NJ4SLBMF).

**Comment by MitchExpensify** [comment]
Sorry to take a step back but does this style of email exist yet? If not is it being introduced here or in another issue?

**Comment by MitchExpensify** [comment]
> So one approach could be to just make text-only adjustments for now, and then follow up and stylize all emails (per David's comment above)

This approach makes sense to me in that case. We should stylize all emails holistically as a first step - Checking if an issue exists for that project [here](https://expensify.slack.com/archives/C05NJ4SLBMF/p1706408763733179?thread_ts=1705345695.801229&cid=C05NJ4SLBMF)

**Comment by MitchExpensify** [comment]
Tracking issue for stylizing emails here: https://github.com/Expensify/Expensify/issues/364323

**Comment by shawnborton** [comment]
Any update here @anmurali? Let me know if you need anything else from me!

**Comment by anmurali** [comment]
Just posted to the thread on this topic [here](https://expensify.slack.com/archives/C05NJ4SLBMF/p1707150811804579?thread_ts=1705345695.801229&cid=C05NJ4SLBMF) but I don't see why we need to punt updating this email for style and wait for one catch all update. We should just update as we go.

**Comment by anmurali** [comment]
I am going to unassign you. @MitchExpensify we need to get an engineer assigned to this.

**Comment by MitchExpensify** [comment]
Cool, I'll hunt down a volunteer!

**Comment by francoisl** [comment]
Discussed [in Slack](https://expensify.slack.com/archives/C05NJ4SLBMF/p1707959431719459?thread_ts=1707951916.206039&cid=C05NJ4SLBMF), going to put this back in the pool to work on other wave issues. I'll grab it back if it gets no love when I have bandwidth.

**Comment by marcaaron** [comment]
I see that if I try to trigger this email locally I hit this block of code:

https://github.com/Expensify/Web-Expensify/blob/90a0948df54c48474cdb781a52d56d9bea004b72/lib/ReportUtils.php#L2569-L2572

@danieldoglas do you have any more context about that? I can disable it locally to get this email to send of course - but unclear whether all users would get moved onto this beta (I think the answer is that we are using it to add people e.g. testers as needed - but figured I'd ask).

**Comment by marcaaron** [comment]
Checked in on this question:

>We want to continue to notify the user immediately via email when they receive a money request
>>Do we do this already today? If so, seems like we can remove this as a requirement. Sending this as a one off means we don’t have to worry about including other messages. Which makes things easier :+1:

And it looks like we don't send it immediately. So, that's the first thing to change here.

**Comment by marcaaron** [comment]
Left a few more questions [here](https://expensify.slack.com/archives/C05NJ4SLBMF/p1707964299101719?thread_ts=1707951916.206039&cid=C05NJ4SLBMF).

**Comment by danieldoglas** [comment]
> (I think the answer is that we are using it to add people e.g. testers as needed - but figured I'd ask).

Yes, we were sending thousands of emails to a few customers and non-used emails (like @expensify.anon) that were making the webrock cluster suffer during certain periods. So we removed those summary emails for them. Here's a bit of context if you're interested: https://expensify.slack.com/archives/C094TGUTZ/p1700331615222279

**Comment by marcaaron** [comment]
Makes sense thanks! I'm gonna disable those checks on DEBUG.

**Comment by marcaaron** [comment]
Ok basic logic for this is done (in the PR I have open) and the notification will look like this (for now):

![image](https://github.com/Expensify/App/assets/32969087/d660b307-3c0a-40e0-bfc5-5907312b1a85)

And the CTA will redirect to NewDot instead of OldDot (like it was before). And also land the user in the parent report vs. the iou report or expense report view (that's where the "pay" button will be in the chat itself).

I'm going to push to get this merged and it should be easy to come back in with a second PR to hook up the styling. Which it sounds like @rlinoz is working on?

**Comment by rlinoz** [comment]
Huum yeah, I wasn't sure if I was supposed to include the styling for money requests in the issue I am working on, but I think this clarifies it.

**Comment by marcaaron** [comment]
Found an interesting edge case for this flow (unsure if we are working on it elsewhere):

**Case 1:**
1. Create a comment on an IOU report or Expense Report in NewDot
2. Wait for the notification to send
3. The email subject is the same as when we make a request - yet there are only comments inside (no requests):

![2024-02-15_11-24-20](https://github.com/Expensify/App/assets/32969087/e8735a42-9063-45e7-9224-a8d5eb3f7b0d)

It's also a bit weird if you do some combination of request + comments e.g. comment then request money...

**Case 2:**
1. Create a comment on an IOU report or Expense Report in NewDot
2. Request money on the same report
3. Wait for the notification to send
4. There is a mixture of comment and request line

![2024-02-15_11-28-08](https://github.com/Expensify/App/assets/32969087/d18dcb5b-23ae-4641-a56f-392ea94450e6)

----

For Case 1 - I think my expectations are that we'd send something that looks more like a report comment notification that just links to the report like this:

![2024-02-15_11-31-39](https://github.com/Expensify/App/assets/32969087/b4dd3680-9823-4785-8f2c-4365649c4170)

For Case 2 - I'm not sure? It feels like we should send one email with the comment summary and another for each individual request?

Does everyone agree with that (I can look into those tweaks now)? Or would we prefer to handle this as a separate issue?

**Comment by marcaaron** [comment]
Though, it does look like having the comments appear under the same email thread is the expected behavior based on the changes [here](https://github.com/Expensify/Web-Expensify/pull/39124).

**Comment by deetergp** [comment]
Yes, those changes were made as a part of the Teachers Unite campaign back in the fall. I don't think they're written in stone, so if something else makes makes more sense going forward, let's entertain it. But loop in @trjExpensify just in case.

**Comment by trjExpensify** [comment]
I haven't been following this issue, so might not be following its history or what's being asked fully, but..

If the proposal is to take any new request out of the "normal" unread summary email and send it as a standalone email with a custom subject header (that links to the parent chat not the expense report), then I would expect we don't also send the unread summary email for the expense report that the request was added to at the same time. 

Outside of that, notifying about "unread" comments added to an expenseReport or an expense thread would be treated like normal: 

- they use the report title seen in product as the subject line i.e Duraflame owes $50.00 (expense report), $25.00 request (expense thread).
- as a result they "thread" in the email client inbox just like any other chat/report
- it includes the comments they missed in the body of the email (one of which might be a request if they haven't read it since it was added to the report) 

Apologies again if I've missed something or what was being asked here, coming in a lil' cold!

**Comment by marcaaron** [comment]
>I would expect we don't also send the unread summary email for the expense report that the request was added to at the same time

@trjExpensify I think you are saying something like this...

- The emails should have the same subject so that threading works correctly
- If a request is made we will send a separate email immediately for that
- If a comment is made we would queue those up and then send them as a summary

In the case where a comment (or multiple) precede a money request we can send two emails. One for the summary and one for the money request. Everything should thread up in the same email chain.

**Comment by marcaaron** [comment]
That _feels like_ what makes the most sense - as they will have different CTA links at the end of the day.

**Comment by trjExpensify** [comment]
>The emails should have the same subject so that threading works correctly

Ah, not quite. I still think a request sends a custom subject email which is "standalone" from the unread messages email thread, we just don't double up and send an unread summary for that chat as well if there aren't any additional messages unread etc. Outside of that, I think we're on the same page:

**Just a request:**
1. UserA is offline
2. UserB sends a request to UserA
3. UserA gets a "custom subject email" with the request immediately
4. No additional "unread summary email" is sent ten minutes later

**Request & messages:**
1. UserA is offline
2. UserB sends a message to UserA 
3. UserB sends a request to UserA
4. UserA gets a "custom subject email" with the request immediately
5. Ten mins later gets the normal "unread summary email" containing both any messages and requests unread in order

**Comment by marcaaron** [comment]
Ok, so we don’t care if the “unread summary” comes out of order? e.g. a comment that came _before_ a request can appear in your inbox _after_ the standalone request? That sounds fine to me, but want to make sure.

**Comment by marcaaron** [comment]
The only technical problem with that is that we have a single unique `NotifyOfflineUsersAboutActivity` job. So we can either:

- Create a new job for the IOU request email (not that hard, but more complex)
- Send only the "money request" action in the `NotifyOfflineUsersAboutActivity` job. Then queue another job for the comments later.

**Comment by marcaaron** [comment]
Heads up, I'm going OOO next week. For that reason - I'm gonna scale back these changes for now so that we are mainly only updating the subject line + sending the summary email immediately (which seems fine for now).

Here's what I think we need to do to achieve @trjExpensify's suggestion above:

1. Create a new job called `SendReportActionNotification` that can be used as a one-off similar to the "custom subject" job. I checked out the custom subject job and it doesn't feel quite right for this as we don't really have a "custom subject" at all. Maybe it can be used, but wasn't immediately obvious to me how and this feels like a different use case.
2. The new job will not be `unique`.
3. It will also not set the `'lastNotified` user nvp (and not affect any already queued `NotifyOfflineUsersAboutActivity` jobs).

**Comment by blimpich** [comment]
> @blimpich, are you already working on a solution to this issue?

Yes and no. https://github.com/Expensify/Expensify/issues/365156 has many similarities. Lets break it down into use cases:

1. user A requests money from user B
2. user A sends money to user B
3. user B completes a payment request from User A
4. user A transfers account balance from wallet to their bank account

From my understanding this issue covers use case number 1. https://github.com/Expensify/Expensify/issues/365156 covers 2, 3, and 4. They are handled differently because number 1 doesn't require a receipt because no money has been exchanged yet. 2, 3, and 4 are all cases where money moved, so we have to follow regulatory requirements and give _a lot_ more information in the email notification.

So @marcaaron and I will probably end up making very similar looking PRs but they will probably need to be separate PRs because of the different requirements.

That being said we should definitely be consistent with the behavior of our summary emails. So to answer this question: 

> so we don’t care if the “unread summary” comes out of order?

I would say yes, we do care. We don't want redundant emails, and the email inbox should be chronological. The [proposed solution](https://expensify.slack.com/archives/C05RECHFBEW/p1708028786579639?thread_ts=1708022278.579259&cid=C05RECHFBEW) that a few of us in #vip-split decided on was to send the current batch of unread messages before sending the custom subject email.

**Comment by trjExpensify** [comment]
>They are handled differently because number 1 doesn't require a receipt because no money has been exchanged yet. 2, 3, and 4 are all cases where money moved, so we have to follow regulatory requirements and give a lot more information in the email notification.

Okay, got it. That's where I thought the "custom footer" part comes into it - but the logic for when a custom subject email is sent for a request would be shared. I.e this part: 

>The [proposed solution](https://expensify.slack.com/archives/C05RECHFBEW/p1708028786579639?thread_ts=1708022278.579259&cid=C05RECHFBEW) that a few of us in #vip-split decided on was to send the current batch of unread messages before sending the custom subject email.

**Comment by marcaaron** [comment]
Quick heads up that raising the PR for what I described in [this comment](https://github.com/Expensify/App/issues/34167#issuecomment-1949387479) could to take me some time as I'm shifting priorities to get the **Group Chats** project shipped ASAP. But, it _sounds like_ that's **not** what we want anymore. @blimpich can you confirm?

What I've got done for this issue so far:

- Update the subject line (PR in review)
- After @rlinoz's changes have been merged we've also got the preview showing now as well 🆒 

**Result:**

![2024-02-26_16-16-38](https://github.com/Expensify/App/assets/32969087/e63a5859-41ec-4d3e-9d83-f450dd29390f)

So - it's getting there... 

----

>send the current batch of unread messages before sending the custom subject email

This is kind of the solution that I started with - if that's how we want to handle things then I don't think we should create a new job at all. It sounds like we just need to add some behavior to "flush" the comments then immediately send another notification for the money request.

>So @marcaaron and I will probably end up making very similar looking PRs but they will probably need to be separate PRs because of the different requirements.

If we can do the main logic for this in one PR that would be good. It sounds like the "flush comments and send money request notification" is the behavior that we want. So, let's set that up for everything. Then come back here after and add the finishing design touches on this one and it can be closed?

**Comment by blimpich** [comment]
That sounds good. I have some of the logic for this figured out already in this [draft PR](https://github.com/Expensify/Web-Expensify/pull/40987/files). You can see I used some of the bits of code from your PR there, so yeah, there is definitely overlap. I made a function that flushes out the unread notifications called `sendQueuedNotificationsForOfflineParticipantsImmediately` which should work for both of our cases.

The only thing is that I'm also not going to be the fastest on this since I'm working my way through the design doc process. It is my main priority but I am partly blocked until I can get the go-ahead on my detailed design. Once that's done my PR should go through pretty quickly and then resolving this issue should be a lot simpler.

So @marcaaron you'll prioritize Group Chats while I work on p2p receipts, which should save you some work when you get freed up next. Sound like a plan?

**Comment by marcaaron** [comment]
@blimpich Sounds pefect. And no worries I don't this one is _super_ urgent.

Lemme know if you need an extra set of eyes on the design doc front 🤙

**Comment by marcaaron** [comment]
Holding this at a lower priority for now.

**Comment by marcaaron** [comment]
Still holding - I am focused on Group Chats. I will try to hand this off.

**Comment by rlinoz** [comment]
As I might pick this up, let me understand a few things.

@trjExpensify based on your comment here: https://github.com/Expensify/App/issues/34167#issuecomment-1948268556

Is this still the behavior we want?

If it is, when are we considering that user A requested money? When the report is submitted, when user A creates a report with an expense, or every new expense created in a report should fire an email?

**Comment by trjExpensify** [comment]
I don't think it is, @rlinoz. The desired behaviour is linked [here](https://github.com/Expensify/App/issues/34167#issuecomment-1953242210). 

>I would say yes, we do care. We don't want redundant emails, and the email inbox should be chronological. The [proposed solution](https://expensify.slack.com/archives/C05RECHFBEW/p1708028786579639?thread_ts=1708022278.579259&cid=C05RECHFBEW) that a few of us in #vip-split decided on was to send the current batch of unread messages before sending the custom subject email.

**Comment by marcaaron** [comment]
@rlinoz are you still interested? I would definitely welcome the help as I am focused on launching Group Chats atm

**Comment by rlinoz** [comment]
The mentions v2 design doc got reviewed in the meantime and I started working on those issues, sorry

**Comment by marcaaron** [comment]
I am blocking this issue here so gonna un-assign myself for now and apply the Hot Pick label. I have a large number of Group Chats issues to follow up on and won't have time to look at the changes above for a while. Sounds like we have a general plan to move forward on so happy to answer any related questions if they come up.

**Comment by trjExpensify** [comment]
What remains here? Is it just implementing the part to use @blimpich's `sendQueuedNotificationsForOfflineParticipantsImmediately` to send any unread comments that might be in and around the request so they’re in chronological order?

**Comment by marcaaron** [comment]
@blimpich what's the state of that PR you linked there? I see it was closed - did we/are we opening a new one?

**Comment by marcaaron** [comment]
I think this is where we're at:

**Proposal:** Send the current batch of unread messages before sending a separate custom email for money requests

**Background:**

When a user requests money from another user we currently:

- Queue an immediate notification (i.e. NotifyOfflineUsersAboutActivity will run as soon as possible)
- The subject line will get updated to read `<Sender> requests <amount>`
- The report link takes you to the `parentReportID` i.e. the “parent” workspace or DM chat of the request
- The “Pay” button takes you directly to the expense report itself

**Problem:**

If there are already unread messages for the user we are notifying those will also display in this same email. This unnecessarily hides comments inside an email that appears to notify you of about one thing.

**Solution:**

- Flush any existing notifications for other report actions first (without the request action)
- Queue a separate notification for the request action with the custom subject and correct report link

Does that sound correct to everyone?

**Comment by trjExpensify** [comment]
That sounds right to me! @anmurali, what about you?

**Comment by blimpich** [comment]
@marcaaron for the solution one clarifying question: does the flushing of existing notifications and queuing of a separate notification for the request action happen immediately upon the user creating the request? It doesn't wait 10 minutes right?

**Comment by marcaaron** [comment]
Correct. It should be immediate.

**Comment by cristipaval** [comment]
I'm ooo this week. This issue will be my top priority next week.

**Comment by cristipaval** [comment]
Making this a daily and starting on it today.

**Comment by cristipaval** [comment]
I'm still working on it. What I found so far is that when a money request is created, the `NotifyOfflineUsersAboutActivity` job runs instantly for the IOU/expense report, not for the parent chat. I need to get the parent chat's `reportID` and call the `sendQueuedNotificationsForOfflineParticipantsImmediately` function for it before.

Right now I'm working on getting the parent chat's `reportID`, but get it the right way, with no additional Auth call.

**Comment by cristipaval** [comment]
I'll do my best to spend a bit of time today on this. Currently working on a #newdot-quality issue.

**Comment by cristipaval** [comment]
^PR up that makes the pending email with the activity to be flushed before the email with the money request

**Comment by cristipaval** [comment]
I'll continue on this one after finishing my #newdot-quality issue from my plate

**Comment by cristipaval** [comment]
I'm writing the detailed section for the Instant Submit for Control plans, which has higher priority

**Comment by cristipaval** [comment]
I'm stepping back from this as I have a critical #newdot-quality issue and other wave-control staff on my plate.

**Comment by trjExpensify** [comment]
@cristipaval it would be helpful if you could provide a quick summary of what's left to do for this issue, now it's back in the pool. I see the Web PR is deployed: https://github.com/Expensify/Web-Expensify/pull/42416

But then, it's unclear what [needs to happen next](https://github.com/Expensify/App/issues/34167#issuecomment-2186766596)? 

>I'll continue on this one after finishing my #newdot-quality issue from my plate

**Comment by cristipaval** [comment]
> I think this is where we're at:
> 
> **Proposal:** Send the current batch of unread messages before sending a separate custom email for new money requests
> 
> **Background:**
> 
> When a user requests money from another user we currently:
> 
> * Queue an immediate notification (i.e. NotifyOfflineUsersAboutActivity will run as soon as possible)
> * The subject line will get updated to read `<Sender> requests <amount>`
> * The report link takes you to the `parentReportID` i.e. the “parent” workspace or DM chat of the request
> * The “Pay” button takes you directly to the expense report itself
> 
> **Problem:**
> 
> If there are already unread messages for the user we are notifying those will also display in this same email. This unnecessarily hides comments inside an email that appears to notify you of about one thing.
> 
> **Solution:**
> 
> * Flush any existing notifications for other report actions first (without the request action)
> * Queue a separate notification for the request action with the custom subject and correct report link
> 
> Does that sound correct to everyone?

This is the PS that we all agreed on, [here](https://github.com/Expensify/App/issues/34167#issuecomment-2123464701).
From the above, we partially implemented the Solution as follows:

> * Flush any existing notifications for other report actions first (without the request action)

This was implemented in [this PR](https://github.com/Expensify/Web-Expensify/pull/42416)

> * Queue a separate notification for the request action with the custom subject and correct report link
We already have a separate email notification for the money requests. We need to audit to make sure it has the correct title and also verify where the Pay button links to: the parent report (policy chat) or the IOU/expense report.

**Comment by trjExpensify** [comment]
>The subject line will get updated to read <Sender> requests <amount>

Marc updated that a while back [here](https://github.com/Expensify/Web-Expensify/pull/40903) I believe. 

>The “Pay” button takes you directly to the expense report itself

Vivek fixed that [here](https://github.com/Expensify/Web-Expensify/pull/42644). 

So this is how it looks:

<img width="1151" alt="image" src="https://github.com/user-attachments/assets/77e711e7-5323-4715-a5ac-4e5faf72c19c">

One caveat, it's now not sending at all because of [this change](https://github.com/Expensify/Auth/pull/11921), but we're talking about how to resolve this [here](https://expensify.slack.com/archives/C036QM0SLJK/p1723490907808029?thread_ts=1722872563.829979&cid=C036QM0SLJK).

**Comment by cristipaval** [comment]
Are we good to close this issue then? 🤔

**Comment by trjExpensify** [comment]
If that's all you were doing as a next step, yep. Go ahead!
